Kudu.Core/Functions/FunctionManager.cs

```diff
@@ -205,9 +205,28 @@ private static bool IsFunctionEnabled
 
         public async Task<JObject> GetHostConfigAsync()
         {
-            return FileSystemHelpers.FileExists(HostJsonPath)
-                ? JObject.Parse(await FileSystemHelpers.ReadAllTextFromFileAsync(HostJsonPath))
-                : new JObject();
+            var host = await TryGetHostConfigAsync();
+            if (host == null)
+            {
+                throw new FileNotFoundException("Host.json is invalid");
```

Comment:
`FileNotFoundException` doesn't seem like the right exception type here since the file is there.